console/src/modules/chatroom/chatroomNotes/useDeleteChatroomNote.ts

```diff
@@ -0,0 +1,30 @@
+import { Reference } from "@apollo/client";
+import { ChatroomNotesDocument, useDeleteChatroomNoteMutation } from "~src/codegen/graphql";
+
+export function useDeleteChatroomNote(chatroomId: string) {
+  return useDeleteChatroomNoteMutation({
+    update(cache, { data }) {
```

Comment:
If this were a real production app, I probably would avoid manually modifying the cache in the case of this mutation, and instead lean into `refetchQueries` with the `awaitRefetchQueries: true` option.

I say that for a couple reasons:

1) It appears that directly modifying the cache does not trigger a re-render. If it were to trigger a re-render and remove the note from the UI, we could avoid returning a promise in `onQueryUpdated` and therefore not have to wait for the list of notes to refetch. Right now, were returning a promise so that the user only returns to the list of notes once the new list, that doesn't include the deleted note, has been refetched.  
2) A bit more risk is introduced when manually modifying the cache, and the code can be more difficult to follow and/or debug. In the case of this mutation, the behavior is actually pretty straightforward, but `useToggleChatroomResolved` is an example where the manual cache update code is a bit less readable and maybe not worth the time and effort.